### PR TITLE
Scripting access of django databases

### PR DESCRIPTION
**Pull request by deming7h777**

### CODE REVIEW

**Comment by hsiaoyi0504** [comment]
Which issue is this PR for ? It seems to me that the title of this PR isn't pretty clear.

**Comment by deming7h777** [comment]
> Which issue is this PR for ? It seems to me that the title of this PR isn't pretty clear.

sorry about that, I change the name of pull request name

**Comment by r06942072** [comment]
[Trouble-shooting needed]
I cloned down, and switch into an empty virtualenv, and start install all dependencies.
I ran into a error when I do 
```pip install -r requirements.txt```

0617_pip_install_rr

**Comment by deming7h777** [comment]
> [Trouble-shooting needed]
> I cloned the project to gmod-dev server, and switch into an empty virtualenv, and start install all dependencies.
> I ran into a error when I do
> `pip install -r requirements.txt`
> 
> 0617_pip_install_rr

don't use server but use your local machine to test again

**Comment by deming7h777** [comment]
> > [Trouble-shooting needed]
> > I cloned the project to gmod-dev server, and switch into an empty virtualenv, and start install all dependencies.
> > I ran into a error when I do
> > `pip install -r requirements.txt`
> > 0617_pip_install_rr
> 
> don't use server but use your local machine to test again, I think that is the environment issue for setting up not for my branch

**Comment by r06942072** [comment]
I met a problem and this is what I did: 
1. Spin up the project by genomics-VM.
https://github.com/NAL-i5K/genomics-VM

2. Enter virtualenv
```cd genomics-workspace```
```source py2.7/bin/activate```

3. ```python manage.py runserver```,  and I got a message as below

0721_vagrant

**Comment by r06942072** [comment]
- postgresql not found
0722_pgsql_notfound

**Comment by r06942072** [comment]
https://github.com/NAL-i5K/genomics-workspace/issues/280

**Comment by hsiaoyi0504** [comment]
@r06942072 As far as I can tell, you should follow the step of setup guide. https://genomics-workspace.readthedocs.io/en/latest/setup_guide_centos.html

**Comment by deming7h777** [comment]
> The module is not found
> This is the process I did
> 
> 1. `git clone --branch addorg2 https://github.com/NAL-i5K/genomics-workspace.git`
> 2. `virtualenv venv` (I create virtualenv myself, but I do not know if there exist one in repo)
> 3. `source venv/bin/activate`
> 4. `pip install -r requirements.txt`
> 5. new added python manage.py command as below screenshot
>    image

try to add __init__.py under blast/management/commands

**Comment by hsiaoyi0504** [comment]
@deming7h777 
If this is the solution to @r06942072 's problem,
> try to add **init**.py under blast/management/commands

maybe it should be added directly on that branch.

File: misc/add_func.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        print("make sure your name is completed and correct")
+        sys.exit(0)
+
+def delete_org(name):
+    #if options["organism"]:
+    #for organism in options["organism"]:
+    #organism = options["organism"][0].lower().capitalize() + " " + options["organism"][1].lower()
+    Organism.objects.filter(display_name = name).delete()
+    return ("remove %s in database"%name)
+'''
```
Comment (hsiaoyi0504) [inline misc/add_func.py]: Don't include these commented out lines. It seems to me that these lines are for initial development, right ?

File: misc/add_func.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
+from blast.models import BlastDb, SequenceType
+#from django.core.management.base import BaseCommand, CommandError
```
Comment (hsiaoyi0504) [inline misc/add_func.py]: Remove unused import directly rather than commenting it out.

File: misc/add_func.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        re1 = requests.get(url1)
+        data1 = re1.json()
+        try:
+            title = data1['query']['search'][0]['title']
+            url2 = wiki_url2 + title
+            re2 = requests.get(url2)
+            data2 = re2.json()
+            key = data1['query']['search'][0]['pageid']
+            key = str(key)
+            #print type(key)
```
Comment (hsiaoyi0504) [inline misc/add_func.py]: remove debug purpose code will be great.

File: misc/add_func.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        data = re.json()
+        tax_id = data['esearchresult']['idlist'][0]
+        tax_id = int(tax_id)
+        return tax_id
+    except IndexError:
+        print("make sure your name is completed and correct")
+        sys.exit(0)
+
+def delete_org(name):
+    #if options["organism"]:
```
Comment (hsiaoyi0504) [inline misc/add_func.py]: Remove unused lines here.

**Comment by r06942072** [comment]
> @r06942072 As far as I can tell, you should follow the step of setup guide. https://genomics-workspace.readthedocs.io/en/latest/setup_guide_centos.html

@hsiaoyi0504  I use repo genomics-VM to set up a new VM, so I didn't go through the process manually in Setup Guide

**Comment by r06942072** [comment]
@deming7h777 
I tried to run those commands on #280 
only ```addorganism``` work

- Here is what I did (I use honeybee as an example organism)
  - ```cd ~/Downloads```
  - ```wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/003/254/395/GCF_003254395.2_Amel_HAv3.1/GCF_003254395.2_Amel_HAv3.1_genomic.fna.gz```
  -  ```gzip -d *.gz```
  - ```cd ~/genomics-workspace```
  - ```python manage.py addorganism Apis mellifera```   (This one work, I checked it through browser)

- Down below three screenshot described the bug I met:
  - 3 commands not found
0725_unknown_3

  - ```addblast```
0725_addblast

  - ```addhmmer```
0725_addhmmer

**Comment by deming7h777** [comment]
@r06942072 you get the error because you don't need to type the full path of fasta file , just the name and title of blast db or hmmer db

**Comment by r06942072** [comment]
> @r06942072 you get the error because you don't need to type the full path of fasta file , just the name and title of blast db or hmmer db

Do you mean like this? I got the same error.
0725_addblast_2
0725_addblast_3



> @r06942072 you get the error because you don't need to type the full path of fasta file , just the name and title of blast db or hmmer db

**Comment by hsiaoyi0504** [comment]
@deming7h777 In my opinion, it will be good to handle path no matter it‘s full path or not. That is, you should handle relative path and absolute path at the same time.

**Comment by r06942072** [comment]
Friday, I pull the update of ```addorg2```, and ran, same error messsage
- ```pull```
0726_pull_addorg2
- ```run```
0726_addblast_makeblast

**Comment by r06942072** [comment]
Monday, what I did
```cd ~/genomics-workspace```

*On personal laptop, make a folder called 'media'
```mkdir -p media/blast/db media/hmmer/db```
```cp ../Downloads/GCF_003254395.2_Amel_HAv3.1_genomic.fna ./media/blast/db/```
```cp ../Downloads/GCF_003254395.2_Amel_HAv3.1_genomic.fna ./media/hmmer/db/```

*command 1 works
```python manage.py addblast Apis mellifera -t peptide protein -f GCF_003254395.2_Amel_HAv3.1_genomic.fna```

*command2 not works
```python manage.py blast_utility GCF_003254395.2_Amel_HAv3.1_genomic.fna -m```
see screenshot

0729_addorg2

**Comment by deming7h777** [comment]
> > Monday, what I did
> > `cd ~/genomics-workspace`
> > *On personal laptop, make a folder called 'media'
> > `mkdir -p media/blast/db media/hmmer/db`
> > `cp ../Downloads/GCF_003254395.2_Amel_HAv3.1_genomic.fna ./media/blast/db/`
> > `cp ../Downloads/GCF_003254395.2_Amel_HAv3.1_genomic.fna ./media/hmmer/db/`
> > *command 1 works
> > `python manage.py addblast Apis mellifera -t peptide protein -f GCF_003254395.2_Amel_HAv3.1_genomic.fna`
> > *command2 not works
> > `python manage.py blast_utility GCF_003254395.2_Amel_HAv3.1_genomic.fna -m`
> > see screenshot
> > 0729_addorg2
> 
> please run `npm run build` first, finish the set up guide first

**Comment by r06942072** [comment]
July.31
- create a checklist for this PR in #280 

- ```sudo yum install npm```
- npm 
0731_node_and_npm

- ```npm run build``` not success, see error message
0731_npm run build  
...
0731_npm_error

- Try to **makeblastdb** on admin page through mouse clicking, but get error message same as running script, **no such file**
0731_addorg2_webpage  
0731_makeblastdb_on_website

- try **addhmmer**, but an error is too few argument
0731_addhmmer

**Comment by r06942072** [comment]
July31, 2nd post
- update npm and node to a higher version, and run ```npm run build```  
0731_success_node_npm

-  successfully run  ```npm run build```  
0731_npm_success


- successfully run  
```python manage.py blast_utility GCF_003254395.2_Amel_HAv3.1_genomic.fna -m```  
```python manage.py blast_utility GCF_003254395.2_Amel_HAv3.1_genomic.fna -p```  
0731_success_m_p

- check those through browser  
0731_success_check


- successfully run   (It was my typo in July31, 1st post)
```python manage.py addhmmer Apis mellifera -f GCF_003254395.2_Amel_HAv3.1_genomic.fna```  

- command ```python manage.py delete``` not found   
0731_unknown_command_delete

**Pull request merged**